projects/plugins/jetpack/extensions/blocks/cookie-consent/cookie-consent.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Cookie-consent Block.
+ *
+ * @since $$next-version$$
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\CookieConsent;
+
+use Automattic\Jetpack\Blocks;
+use Jetpack_Gutenberg;
+
+const FEATURE_NAME = 'cookie-consent';
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+const COOKIE_NAME  = 'eucookielaw';
+
+/**
+ * If the site has ads, it will have its own cookie widget. This function checks if the site has ads.
+ *
+ * @since $$next-version$$
+ *
+ * @return bool
+ */
+function should_use_ads_cookie_widget_instead() {
+	return ! ( ( defined( 'NOADVERTS' )
+	|| defined( 'NOADSUPGRADE' )
+	|| 1 == get_option( 'permanent_noadverts' ) // phpcs:ignore Universal.Operators.StrictComparisons.LooseEqual,WordPress.PHP.StrictComparisons.LooseComparison -- This can be stored as string as well.
+	|| ( function_exists( 'has_blog_sticker' ) && has_blog_sticker( 'wordads' ) ) ) ); // Enable widget for WordAds users (regardless of plan).
+}
```

Comment:
Will this work on self-hosted sites and WoA sites? That `permanent_noadverts` option is not available on self-hosted sites, for example.